prow/plugins/releasenote/releasenote.go

```diff
@@ -235,11 +231,15 @@ func handlePR(gc githubClient, log *logrus.Entry, pr *github.PullRequestEvent) e
 		log.Error(err)
 	}
 
-	return clearStaleComments(gc, log, pr, prLabels, comments)
+	return clearStaleComments(gc, log, pr, comments)
 }
 
-func clearStaleComments(gc githubClient, log *logrus.Entry, pr *github.PullRequestEvent, prLabels []github.Label, comments []github.IssueComment) error {
-	// Clean up old comments.
+// Clean up old comments.
+func clearStaleComments(gc githubClient, log *logrus.Entry, pr *github.PullRequestEvent, comments []github.IssueComment) error {
+	prLabels, err := gc.GetIssueLabels(pr.Repo.Owner.Login, pr.Repo.Name, pr.Number)
```

Comment:
Instead of using another API token to fetch the labels again to get the label set after labels are added and removed, we should modify `prLabels` when we add or remove a label to maintain the current state and then reuse it here.
This will be cleaner if you change `prLabels` to a `map[string]bool`.

Reply:
Agree on reusing updated prLabels instead of using another API token. But I'm not sure if I want to change the type of prLabels. This will lead to changing a lot of unrelated codes.

Reply:
>I'm not suggesting that you change the signature of GetIssueLabels, I'm suggesting that you use the result of GetIssueLabels to build a set of labels that is used from then on in the releasenote plugin and kept up to date by adding and removing labels from it as needed

Yeah I got your point on this :) What I meant is that this would lead to changing a lot of other `prLabels` related codes, such as `prMustFollowRelNoteProcess()`, `releaseNoteAlreadyAdded()`, `hasLabel()`, etc. I will try to refactor the codes though.